reject.ts

```diff
@@ -0,0 +1,10 @@
+import curryN from "./utils/curry_n.ts"
+import { Func, Curry2, Obj } from "./utils/types.ts"
+import complement from './complement.ts';
+import filter from './filter.ts';
+
+function reject(pred: Func, filterable: Array<any> | Obj) {
```

Comment:
`pred` is applied on one item and returns boolean, right?
So use `Predicate1` defined in `utils/types.ts`
Also use generic type for the array